cmake builds are broken after HTTP compression feature merged

Seems the cmake build finds and uses the system zlib headers but does not link against the proper library.

# syslog-ng
## Version of syslog-ng
master, latest

## Platform
ARM macOS


# Issue
## Failure
Undefined symbols for architecture arm64:
  "_deflate", referenced from:
      __deflate_type_compression_method in compression.c.o
     (maybe you meant: _deflate_compressor_new, __deflate_type_compression , __deflate_compressor_compress )
  "_deflateEnd", referenced from:
      __deflate_type_compression_method in compression.c.o
  "_deflateInit2_", referenced from:
      __deflate_type_compression_method in compression.c.o
ld: symbol(s) not found for architecture arm64

--- a/cmake/print_config_summary.cmake
+++ b/cmake/print_config_summary.cmake
@@ -47,54 +47,75 @@ function(_print_summary_line _variableName _maxVarNameLen)
   endif()
 endfunction()
 
-# Lists actual module definition results
-# Behaves similar to the cmake -L option but filters out the relevant information and adds some highlighting
-#
-function(print_config_summary)
-  list(APPEND _importantVariableNames "IVYKIS_INTERNAL" "BUILD_TESTING")
-  get_cmake_property(_variableNames VARIABLES)
-  list(SORT _variableNames)
-
+function(_print_separator)
   message(NOTICE "-----------------------------------")
-  message(NOTICE "syslog-ng Open Source Edition ${SYSLOG_NG_VERSION} configured")
-  message(NOTICE "-----------------------------------")
+endfunction()
 
-  if(SUMMARY_VERBOSE)
-    foreach(_variableName ${_variableNames})
-      string(REGEX MATCH ".*(_LIBRARY|_LIBRARIES)$" _match "${_variableName}")
-      string(REGEX MATCH "^(CMAKE_|_).*" _cmakeInternal "${_variableName}")
+function(_print_libraries _variableNames)
+  foreach(_variableName ${_variableNames})
+    string(REGEX MATCH ".*(_LIBRARY|_LIBRARIES)$" _match_lib "${_variableName}")
+    string(REGEX MATCH ".*(INCLUDE_DIR|INCLUDEDIR|INCLUDE_DIRS)$" _match_incl "${_variableName}")
+    string(REGEX MATCH "^(CMAKE_|_).*" _cmakeInternal "${_variableName}")
 
-      if(_match AND NOT _cmakeInternal)
-        _print_summary_line(${_variableName} 40)
-      endif()
-    endforeach()
-  endif()
+    if((_match_lib OR _match_incl) AND NOT _cmakeInternal)
+      _print_summary_line("${_variableName}" 40)
+    endif()
+  endforeach()
+endfunction()
 
-  message(NOTICE "-----------------------------------")
-
+function(_print_importants _variableNames _importantVariableNames)
   foreach(_variableName ${_variableNames})
     foreach(_importantVariableName ${_importantVariableNames})
       string(REGEX MATCH "^${_importantVariableName}$" _match "${_variableName}")
 
       if(_match)
-        _print_summary_line(${_importantVariableName} 32)
+        _print_summary_line("${_importantVariableName}" 32)
       endif()
     endforeach()
   endforeach()
+endfunction()
 
-  message(NOTICE "-----------------------------------")
-
+function(_print_options _variableNames)
   foreach(_variableName ${_variableNames})
     string(REGEX MATCH "^ENABLE_" _match "${_variableName}")
 
     if(_match)
-      _print_summary_line(${_variableName} 32)
-    else()
-      if(SUMMARY_FULL)
-        message("${_variableName}=${${_variableName}}")
-      endif()
+      _print_summary_line("${_variableName}" 32)
     endif()
   endforeach()
+endfunction()
 
-  message(NOTICE "-----------------------------------")
+function(_print_full _variableNames)
+  foreach(_variableName ${_variableNames})
+    message("${_variableName}=${${_variableName}}")
+  endforeach()
+endfunction()
+
+# Lists actual module definition results
+# Behaves similar to the cmake -L option but filters out the relevant information and adds some highlighting
+#
+function(print_config_summary)
+  get_cmake_property(_variableNames VARIABLES)
+  list(SORT _variableNames)
+
+  _print_separator()
+  message(NOTICE "syslog-ng Open Source Edition ${SYSLOG_NG_VERSION} configured")
+  _print_separator()
+
+  if(SUMMARY_FULL)
+    _print_full("${_variableNames}")
+  else()
+    if(SUMMARY_VERBOSE)
+      _print_libraries("${_variableNames}")
+      _print_separator()
+    endif()
+
+    list(APPEND _importantVariableNames "IVYKIS_INTERNAL" "BUILD_TESTING")
+    _print_importants("${_variableNames}" "${_importantVariableNames}")
+    _print_separator()
+
+    _print_options("${_variableNames}")
+  endif()
+
+  _print_separator()
 endfunction()

--- a/modules/http/CMakeLists.txt
+++ b/modules/http/CMakeLists.txt
@@ -45,7 +45,9 @@ add_module(
   TARGET http
   GRAMMAR http-grammar
   INCLUDES ${Curl_INCLUDE_DIR}
+           ${ZLIB_INCLUDE_DIRS}
   DEPENDS ${Curl_LIBRARIES}
+          ${ZLIB_LIBRARIES}
   SOURCES ${HTTP_DESTINATION_SOURCES}
 )
 

